x-pack/plugin/ml/src/main/java/org/elasticsearch/xpack/ml/inference/adaptiveallocations/AdaptiveAllocationsScalerService.java

```diff
@@ -261,7 +250,7 @@ public synchronized void start() {
     public synchronized void stop() {
         clusterService.removeListener(this);
         stopScheduling();
-        metrics.close();
+        scalers.clear();
```

Comment:
I've leave the metrics list as is and don't recreate them (which leads to errors).

Clearing the `scalers` collection, so that the metrics do not observe anything anymore.